state/workingset.go

```diff
@@ -178,11 +180,16 @@ func (ws *workingSet) RunActions(
 			return hash.ZeroHash32B, errors.Wrap(err, "failed to convert candidate list to map of cached Candidates")
 		}
 	}
+	// check producer
+	producer, err := ws.LoadOrCreateAccountState(producerAddr, big.NewInt(0))
```

Comment:
You need to also put state of the producer after calculating his incoming

Reply:
it's already handled in stateCandidate